### ISSUE TITLE
site_reviews shortcode display, pagination issues with Elementor

### ISSUE THREAD

**Issue by LoganStrike** [pryley/site-reviews#27]
http://brewtheme1elementor.flywheelsites.com/reviews/

This site has 4 reviews total. The pagination is showing at the bottom of the page even though all reviews are already displayed on the page. Clicking to page 2 shows no reviews as you might expect, but there should be no pagination to begin with.

This is the shortcode on the page, which is inside an Elementor shortcode widget.

`[site_reviews display="10" pagination="true" hide="date"]`

-----
http://brewtheme1elementor.flywheelsites.com/

I also have an issue where the home page is not showing all four reviews even though the shortcode is set to display 4. In addition, while not happening at the moment, I've also had an issue where the pagination shows up on the home page even when pagination shortcode attribute is set to false.

`[site_reviews display="4" hide="date"]`

I'm certain this is not a cache issue as there are no caching plugins being used, and the server cache was flushed via the web hosting interface. Using Site reviews v5.13.2

**Comment by BenPam** [comment]
Same here, 5 pages empty with the "There are no reviews yet. Be the first one to write one." and the last with 7 reviews. It looks like when the plugin query the reviews, it gets reviews and some other type of post which can't be display

**Comment by pryley** [comment]
Have you tried running the Repair Review Relations tool?

**Comment by LoganStrike** [comment]
> Have you tried running the Repair Review Relations tool?

No, it says "Repair is unnecessary because your database tables use the InnoDB engine! ✨"

**Comment by pryley** [comment]
Did you previously at some point run the Optimise Your Database Tables tool to convert tables from MyISAM to InnoDB, or have your database tables always used InnoDB?

**Comment by LoganStrike** [comment]
> Did you previously at some point run the Optimise Your Database Tables tool to convert tables from MyISAM to InnoDB, or have your database tables always used InnoDB?

I never ran that tool. DB has always been InnoDB to my knowledge.

**Comment by pryley** [comment]
@LoganStrike It would help me if you would use the Contact Support section on the Help page to contact me directly and send the Console Log (if applicable) and the Site Reviews System Info. 

@BenPam This sounds like a different issue. Please try running the Repair Review Relations tool, and then run the Optimise Your Database Tables tool.

**Comment by pryley** [comment]
@LoganStrike 

The System Info you sent shows that you have seven 0-star reviews and four 5-star reviews (11 reviews total).

Is this still true? 

![image](https://user-images.githubusercontent.com/134939/132376864-44647bbf-06df-440c-9f1a-b636988adf7d.png)

The reason I ask is because I am trying to reproduce your problem on a local site. Site Reviews will display all 0-5 star reviews by default, but your site shows only the four 5-star reviews.

**Comment by pryley** [comment]
@LoganStrike  Next, if you have access to PHPMyAdmin, TablePlus, or SequelPro (or any other MySQL client), would you be able to run the following SQL query and post a screenshot of the results?

```mysql
SELECT
    r.ID,
    r.review_id,
    r.rating,
    r.is_approved,
    r.is_pinned, 
    p.post_author, 
    p.post_date, 
    p.post_title, 
    p.post_status, 
    p.post_type
FROM wp_glsr_ratings AS r 
INNER JOIN wp_posts AS p ON r.review_id = p.ID 
ORDER BY p.post_date desc
``` 

Note: If you are using a custom table prefix, make sure to update the table names in the query (e.g. `wp_` is the default WordPress table prefix).

For example, here is a screenshot of what it looks like when I run that query:

![image](https://user-images.githubusercontent.com/134939/132381912-2d3e0ff2-9c69-4975-9e9e-76b8b3e97cde.png)

**Comment by pryley** [comment]
@BenPam 

1. Did you ever run the "Optimise Your Database Tables" tool previously?
2. Did you import the database (or any of the database tables) from a different WordPress website?

Here is something to try, please let me know if it works:

You will need to edit the following file in the plugin:

`/wp-content/plugins/site-reviews/views/pages/tools/general/repair-review-relations.php`

Replace all text in that file with the following code: https://pastebin.com/iYzAi3er

Once you have done that, you should be able to run the "Repair Review Relations" tool.

**Comment by LoganStrike** [comment]
@pryley There are only 4 reviews total. When I search the posts table for site-review post type, there are only 4 rows returned there as well.

Here is the result pf the query:

<img width="887" alt="Screen Shot 2021-09-07 at 12 14 09 PM" src="https://user-images.githubusercontent.com/16969785/132398694-15d47e86-db41-4baf-8559-5fbf17b51295.png">

**Comment by pryley** [comment]
@LoganStrike Please see question no.2 in the [reply above](https://github.com/pryley/site-reviews/issues/27#issuecomment-914552529) and the proposed solution.

**Comment by LoganStrike** [comment]
> @LoganStrike Please see question no.2 in the [reply above](https://github.com/pryley/site-reviews/issues/27#issuecomment-914552529) and the proposed solution.

This site wasn't imported from anywhere. It started as a default Flywheel WordPress installation, and the 4 reviews were added by a logged-in administrator submitting each review via the front end form using test@example.com as the email for all of them.

This site will be used to create other WordPress installations using the Flywheel Blueprint feature, which is basically a saved WordPress instance that can be easily duplicated to use as a starting point for other sites. 

The proposed solution seems to have fixed the issue. Do you know what caused it to happen in the first place? Do you anticipate that I'd need to do this repair process on all sites moving forward?

**Comment by pryley** [comment]
I think I found the culprit. There was a post_type check missing when listening for the transition_post_status hook.

You can either wait for the next update which will fix this, or rollback to v5.10.4

**Comment by pryley** [comment]
If you would like to stay on the latest version and patch the problem yourself, please see: https://github.com/pryley/site-reviews/commit/ca3332029898755d76b6d1818bd1d4927bafe64d
